python_modules/dagster/dagster_tests/general_tests/asset_graph_differ_tests/test_asset_graph_differ.py

```diff
@@ -251,3 +251,51 @@ def test_new_code_location(instance):
     assert differ.get_changes_for_asset(AssetKey("new_asset")) == [ChangeReason.NEW]
     assert differ.get_changes_for_asset(AssetKey("upstream")) == [ChangeReason.NEW]
     assert differ.get_changes_for_asset(AssetKey("downstream")) == [ChangeReason.NEW]
+
+
+def test_change_partitions_definitions(instance):
```

Comment:
could expand on this set of tests to be more exhaustive about:
- changing the type of the partition def, not just the start date/set of static partitions
- modifying the internal defs of the multi partition
- adding partitions to a non partitioned asset
- adding partition mapping to an input
- removing partition def or partition mapping

Reply:
since we are comparing "value objects" (trees of namedtuples) I feel like we cover a lot of ground out of the box 

in that light it may be worth prioritizing finding the right high level tuples to compare to expand the total coverage as efficiently as possible